File: JAVA/pertemuan10/switchcase.java
// package declaration removed

import java.util.Scanner;

public class switchcase {
     public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);


        String nama;
        while (true) {
            System.out.print("Masukkan nama Anda: ");
            nama = sc.nextLine();

            System.out.print("Apakah nama sudah benar? (Y/N): ");
            String benar = sc.nextLine().toLowerCase();

            if (benar.equals("y")) {
                break;
            } else {
                System.out.println("Nama salah. Silakan masukkan nama lagi.\n");
            }
        }

        System.out.print("Masukkan NIM Anda Untuk mengecek saldo: ");
        int nim = sc.nextInt();
        double saldo = nim;

        System.out.println("\nHalo, " + nama + "!");
        System.out.println("Saldo Anda adalah: Rp " + saldo);

        // Menu
        System.out.println("""
                ===== Menu =====
                1. Cek Saldo
                2. Tarik Tunai
                3. Setor Tunai
                4. Transfer
                5. Keluar
                """);

        System.out.print("Pilih menu (1-5): ");
        int pilihan = sc.nextInt();

        switch (pilihan) {
            case 1:
                System.out.println(nama + ",");
                System.out.println("Saldo Anda adalah: Rp " + saldo);
                break;

            case 2:
                System.out.print("Masukkan jumlah tarik tunai: Rp ");
                int tarik = sc.nextInt();

                if (tarik > saldo) {
                    System.out.println("Saldo tidak mencukupi untuk tarik tunai sebesar Rp " + tarik);
                } else {
                    saldo -= tarik;
                    System.out.println("Tarik tunai sebesar Rp " + tarik + " berhasil.");
                    System.out.println("Sisa saldo Anda: Rp " + saldo);
                }
                break;

            case 3:
                System.out.print("Masukkan jumlah setor tunai: Rp ");
                int setor = sc.nextInt();

                saldo += setor;
                System.out.println("Setor tunai sebesar Rp " + setor + " berhasil.");
                System.out.println("Saldo Anda sekarang: Rp " + saldo);
                break;

            case 4:
                System.out.print("Masukkan jumlah transfer: Rp ");
                int transfer = sc.nextInt();

                saldo -= transfer;
                System.out.println("Transfer sebesar Rp " + transfer + " berhasil.");
                System.out.println("Saldo Anda sekarang: Rp " + saldo);
                break;

            case 5:
                System.out.println("Keluar dari menu. Sampai jumpa lagi!");
                break;

            default:
                System.out.println("Pilihan tidak valid. Silakan pilih menu antara 1-5.");
                break;
        }

        System.out.println("\nTerima kasih telah menggunakan layanan kami!");
        System.out.println("==============");
 
    }
}

File: C++/pertemuan10/switchcase.cpp
#include <iostream>
#include <string>
#include <iomanip> 

using namespace std;

int main() {
    string nama;
    char benar;

    
    do {
        cout << "Masukkan nama Anda: ";
        getline(cin, nama);

        cout << "Apakah nama sudah benar? (Y/N): ";
        cin >> benar;
        benar = tolower(benar);

        cin.ignore(); 
        if (benar != 'y') {
            cout << "Nama salah. Silakan masukkan nama lagi.\n\n";
        }

    } while (benar != 'y');

    int nim;
    cout << "Masukkan NIM Anda Untuk mengecek saldo: ";
    cin >> nim;
    double saldo = nim;

    cout << "\nHalo, " << nama << "!\n";
    cout << "Saldo Anda adalah: Rp " << saldo << endl;

    cout << "\n===== Menu =====\n"
         << "1. Cek Saldo\n"
         << "2. Tarik Tunai\n"
         << "3. Setor Tunai\n"
         << "4. Transfer\n"
         << "5. Keluar\n\n";

    cout << "Pilih menu (1-5): ";
    int pilihan;
    cin >> pilihan;

    switch (pilihan) {
        case 1:
            cout << nama << ",\n";
            cout << "Saldo Anda adalah: Rp " << saldo << endl;
            break;

        case 2: {
            int tarik;
            cout << "Masukkan jumlah tarik tunai: Rp ";
            cin >> tarik;

            if (tarik > saldo) {
                cout << "Saldo tidak mencukupi untuk tarik tunai sebesar Rp " << tarik << endl;
            } else {
                saldo -= tarik;
                cout << "Tarik tunai sebesar Rp " << tarik << " berhasil.\n";
                cout << "Sisa saldo Anda: Rp " << saldo << endl;
            }
            break;
        }

        case 3: {
            int setor;
            cout << "Masukkan jumlah setor tunai: Rp ";
            cin >> setor;

            saldo += setor;
            cout << "Setor tunai sebesar Rp " << setor << " berhasil.\n";
            cout << "Saldo Anda sekarang: Rp " << saldo << endl;
            break;
        }

        case 4: {
            int transfer;
            cout << "Masukkan jumlah transfer: Rp ";
            cin >> transfer;

            saldo -= transfer;
            cout << "Transfer sebesar Rp " << transfer << " berhasil.\n";
            cout << "Saldo Anda sekarang: Rp " << saldo << endl;
            break;
        }

        case 5:
            cout << "Keluar dari menu. Sampai jumpa lagi!\n";
            break;

        default:
            cout << "Pilihan tidak valid. Silakan pilih menu antara 1-5.\n";
            break;
    }

    cout << "\nTerima kasih telah menggunakan layanan kami!\n";
    cout << "==============\n";

    return 0;
}
